soda_machine.rb

```diff
@@ -6,13 +6,33 @@ def initialize(args = {})
     @cash = args[:cash]
   end
 
+  #the sodas are initialized as an array, so I have to simply find the
+  #length of the array containing the soda objects
   def current_inventory_count
+    @sodas.length
   end
 
+  #returns the soda object that the user request
+  #which are stored in the array of sodas, so I have to search the array for
+  #the soda object
   def find_soda(soda_brand)
+    @sodas.each do |soda|
+      if soda.brand == soda_brand
+        return soda
+      end
+    end
+    return nil
```

Comment:
Look into the `Enumerabe#find` method (http://ruby-doc.org/core-1.9.3/Enumerable.html#method-i-find).  Converts this into a one-liner.